spacy/cli/project/assets.py

```diff
@@ -32,25 +33,42 @@ def project_assets_cli(
     DOCS: https://spacy.io/api/cli#project-assets
     """
     overrides = parse_config_overrides(ctx.args)
-    project_assets(project_dir, overrides=overrides, sparse_checkout=sparse_checkout)
+    project_assets(
+        project_dir,
+        overrides=overrides,
+        sparse_checkout=sparse_checkout,
+        download_all=download_all,
+    )
 
 
 def project_assets(
     project_dir: Path,
     *,
     overrides: Dict[str, Any] = SimpleFrozenDict(),
     sparse_checkout: bool = False,
+    download_all: bool = False,
 ) -> None:
     """Fetch assets for a project using DVC if possible.
 
     project_dir (Path): Path to project directory.
+    sparse_checkout (bool): Use sparse checkout for assets provided via Git, to only check out and clone the files
+                            needed.
+    download_all (bool): Whether to download all assets, including those marked as optional.
     """
     project_path = ensure_path(project_dir)
     config = load_project_config(project_path, overrides=overrides)
-    assets = config.get("assets", {})
+    assets = [
+        asset
+        for asset in config.get("assets", [])
+        if download_all or not asset.get("optional", False)
```

Comment:
Let's move this default to somewhere up in the file so we can see/change it easily if that would ever come up in the future? Something like `OPTIONAL_DEFAULT`.